### ISSUE TITLE
[Enhancement] Add target-mean exposure scaling for low-light augmentation

### ISSUE THREAD

**Issue by ternaus** [albumentations-team/AlbumentationsX#375]
Two independent low-light pipelines choose a desired mean intensity and derive a multiplicative gain from the current image as `target_mean / current_mean`. AlbumentationsX can sample a brightness factor, but it cannot sample a desired mean and compute the input-dependent gain needed to reach it. A native transform would keep this operation inside `Compose`, make it serializable and replayable, and define its batch behavior.

## Public implementations

The documented LCIM training path in [`Vision-and-Multimodal-Intelligence-Lab/UDAPose`](https://github.com/Vision-and-Multimodal-Intelligence-Lab/UDAPose) already depends on `albumentationsx~=2.0.10`. Its dataset runs the geometric pipeline, leaves `Compose`, samples a target mean between `0.3 * 255` and `0.5 * 255`, and rescales the tensor manually ([source](https://github.com/Vision-and-Multimodal-Intelligence-Lab/UDAPose/blob/638d77bf45e7b0a4b5efc2fea19a928a9e100480/data-pipeline/dataset/TrainDataset.py#L92-L139)):

```python
self.transforms = A.Compose(
    [
        A.SmallestMaxSize(512),
        A.RandomCrop(320, 512),
        A.HorizontalFlip(),
        A.VerticalFlip(),
        A.ToTensorV2(),
    ]
)

img = self.transforms(image=img)["image"]
factor = random.uniform(0.3, 0.5)
img = img.float()
img = img * (factor * 255.0 / img.mean())
img = img.round().clamp(0, 255).to(torch.uint8)
```

The same repository uses a fixed target of `0.346 * 255` when converting model outputs for evaluation ([batched source](https://github.com/Vision-and-Multimodal-Intelligence-Lab/UDAPose/blob/638d77bf45e7b0a4b5efc2fea19a928a9e100480/data-pipeline/eval_vae.py#L143-L149), [single-image source](https://github.com/Vision-and-Multimodal-Intelligence-Lab/UDAPose/blob/638d77bf45e7b0a4b5efc2fea19a928a9e100480/data-pipeline/train_vae.py#L329-L333)):

```python
rgb_mean = einops.reduce(img, "b c h w -> b 1 1 1", "mean")
img = img * (0.3460 * 255.0 / rgb_mean)
img = img.round().clamp(0, 255).to(torch.uint8)
```

An independent low-light pipeline in [`Wayneee42/3dv-lowlight`](https://github.com/Wayneee42/3dv-lowlight) calls the same operation `exposure_match`, adds optional gain bounds, and combines it with gamma augmentation ([source](https://github.com/Wayneee42/3dv-lowlight/blob/ce800310dc48d9c1268abb6cc58efbae990f9bac/core/libs/augment.py#L23-L27)):

```python
def exposure_match(image, target_mean, min_scale=1.0, max_scale=3.0, eps=1e-6):
    image = torch.clamp(image, 0.0, 1.0)
    current_mean = image.mean().clamp_min(eps)
    scale = torch.clamp(
        torch.tensor(target_mean, device=image.device) / current_mean,
        min_scale,
        max_scale,
    )
    return torch.clamp(image * scale, 0.0, 1.0), float(scale.item())
```

Its training path samples the target and applies the derived gain ([source](https://github.com/Wayneee42/3dv-lowlight/blob/ce800310dc48d9c1268abb6cc58efbae990f9bac/core/libs/augment.py#L587-L670)):

```python
if training:
    jitter = random.uniform(-target_mean_jitter, target_mean_jitter)
    target_mean = min(max(base_target_mean * (1.0 + jitter), 0.05), 0.95)
else:
    target_mean = base_target_mean

if mode in ("exposure_match", "hybrid"):
    supervision, scale = exposure_match(
        supervision,
        target_mean,
        min_scale=min_scale,
        max_scale=max_scale,
    )
```

## Why the current transforms do not express this

[`RandomBrightnessContrast`](https://albumentations.ai/explore/transform/RandomBrightnessContrast/) samples a multiplicative brightness factor independently of the input mean, so two inputs with different means remain different after receiving the same sampled factor. The code above instead samples the desired output mean and derives a different gain for each input.

[`Normalize`](https://albumentations.ai/explore/transform/Normalize/) with `normalization="image"` subtracts the image mean and divides by its standard deviation, producing a zero-centered float output. [`AutoContrast`](https://albumentations.ai/explore/transform/AutoContrast/) derives its mapping from the image extrema. Neither preserves the source ratios while scaling the global mean toward a sampled target.

## Proposed contract

The proposed API name is `ExposureMatching`; the sampled `target_mean_range` carries the stochastic policy, following the AlbumentationsX convention that new transform names do not use a `Random` prefix:

```python
A.ExposureMatching(
    target_mean_range=(0.3, 0.5),
    gain_range=None,
    p=1.0,
)
```

For dtype maximum `M`, sampled normalized target `t`, and normalized global mean `mu = mean(image) / M`:

```text
gain = t / max(mu, eps)
gain = clip(gain, gain_range)  # when gain_range is provided
output = clip(image * gain, 0, M)
```

`target_mean_range` should use dtype-relative values in `[0, 1]`, so one configuration works for `uint8` and `float32`. `gain_range=(1.0, 3.0)` would reproduce the bounded independent implementation; `None` would reproduce UDAPose. A zero image remains zero because multiplicative exposure cannot create signal.

Clipping saturated pixels or constraining the gain can make the achieved post-clipping mean lower than the sampled target. The first version should preserve the observed one-pass formula and document this limit; iterative saturation compensation is a separate policy.

With the proposed transform, the active UDAPose training path becomes:

```diff
 self.transforms = A.Compose(
     [
-        A.SmallestMaxSize(512),
-        A.RandomCrop(320, 512),
-        A.HorizontalFlip(),
-        A.VerticalFlip(),
+        A.SmallestMaxSize(max_size=512),
+        A.RandomCrop(height=320, width=512),
+        A.HorizontalFlip(p=0.5),
+        A.VerticalFlip(p=0.5),
+        A.ExposureMatching(
+            target_mean_range=(0.3, 0.5),
+            p=1.0,
+        ),
         A.ToTensorV2(),
     ]
 )

 img = self.transforms(image=img)["image"]
-factor = random.uniform(0.3, 0.5)
-img = img.float()
-img = img * (factor * 255.0 / img.mean())
-img = img.round().clamp(0, 255).to(torch.uint8)
```

## Acceptance criteria

- Preserve `uint8` and `float32` dtype and documented ranges.
- Compute the global mean across all pixels and channels, matching both public implementations.
- Sample one target per transform call; for `images` and `volume`, reuse that target while deriving a gain independently for each image or slice.
- Record the sampled target and applied gain or gains through `save_applied_params`, `get_applied_params()`, and `ReplayCompose`.
- Serialize both ranges and reproduce replay exactly on the same input.
- Test zero and near-zero means, saturation, optional gain bounds, arbitrary channel counts, image batches, 3D data, and both supported dtypes.
- Add a parameter-sensitivity benchmark because computing an image mean adds a reduction and a float multiply pass.

GitHub search found no existing open or closed AlbumentationsX issue for target-mean exposure scaling. The two implementations above are independent and use the operation in low-light training or evaluation paths.
